Source/CBL_Router+Handlers.m

```diff
@@ -618,9 +618,15 @@ - (CBLStatus) do_GET: (CBLDatabase*)db docID: (NSString*)docID {
             if (rev)
                 rev = [self applyOptions: options toRevision: rev status: &status];
             if (!rev) {
-                if (status == kCBLStatusDeleted)
-                    _response.statusReason = @"deleted";
-                else
+                if (status == kCBLStatusDeleted) {
+                    if (revID) {
+                        status = kCBLStatusOK;
```

Comment:
I don't think this line is reachable. If -getDocumentWithID:revisionID:withBody:status: is called with a non-nil revID, and it finds a deleted revision, it will return that revision. It will only return nil if there is no such revision, and in that case status will be set to kCBLStatusNotFound.

Reply:
You are right. It turns out iOS doesn't have this issue. I will close the PR and commit only unit test. @borrrden and @hideki, can you review the issue / fix again?